### PR TITLE
Bump iOS/tvOS minimum version to 11.0, cleanup some code

### PR DESCRIPTION
**Pull request by akoeplinger**
In .NET 8 the minimum Apple OS versions we're supporting are iOS/tvOS 11.0 and macOS 10.15.

This bumps the version settings in various files and cleans up code that can now assume it's running on a later OS version. For example, it allows us to get rid of a couple `dlsym()` calls in System.Security.Cryptography.Native.Apple.

We can also stop building the iOS arm and iOSSimulator x86 runtime packs since 32bit platforms aren't supported by iOS 11

### CODE REVIEW

File: src/mono/Directory.Build.props [hunk 1]
```diff
@@ -13,11 +13,11 @@
 
   <PropertyGroup>
     <!-- Minimum target OS version, keep in sync with src/native/libs/build-native.sh -->
-    <iOSVersionMin>10.0</iOSVersionMin>
-    <tvOSVersionMin>10.0</tvOSVersionMin>
+    <iOSVersionMin>11.0</iOSVersionMin>
```
Comment (marek-safar) [inline src/mono/Directory.Build.props:16]: nit: should this move level up because of nativeaot?

Comment (akoeplinger) [reply src/mono/Directory.Build.props:16]: what do you mean? this is unrelated to nativeaot.

Comment (marek-safar) [reply src/mono/Directory.Build.props:16]: I know just commenting on the code as you touch it. I think it'd be better under src/

Comment (akoeplinger) [reply src/mono/Directory.Build.props:16]: ah, yeah that might be a good idea

File: src/mono/cmake/configure.cmake [hunk 2]
```diff
@@ -99,9 +99,9 @@ endif()
 
 check_include_files("sys/types.h;sys/user.h" HAVE_SYS_USER_H)
 
-if(NOT HOST_DARWIN)
-  # getentropy was introduced in macOS 10.12 / iOS 10.0
-  ac_check_funcs (getentropy)
+if(HOST_IOS OR HOST_MACCAT)
+  # getentropy isn't allowed in the AppStore: https://github.com/rust-lang/rust/issues/102643
+  set(HAVE_GETENTROPY 0)
```
Comment (filipnavara) [inline src/mono/cmake/configure.cmake:104]: If it's not allowed on the AppStore then I think we should keep excluding it for macOS/tvOS too (even if macOS is not supported config for Mono anymore).

**Review by filipnavara — APPROVED** [review]
LGTM

File: src/mono/cmake/configure.cmake (hunk 2 above)
Comment (akoeplinger) [reply src/mono/cmake/configure.cmake:104]: Yeah one thing that might be confusing is that HOST_IOS includes tvOS in the mono cmake build system. Longer term I'd like to unify differences like this with the rest of the repo.

File: src/mono/Directory.Build.props (hunk 1 above)
Comment (akoeplinger) [reply src/mono/Directory.Build.props]: Done

**Review by marek-safar — APPROVED** [review]

**Pull request merged**